redis/commands.cc

```diff
@@ -129,6 +129,26 @@ future<redis_message> echo::execute(service::storage_proxy&, redis::redis_option
     return redis_message::make_strings_result(std::move(_str));
 }
 
+shared_ptr<abstract_command> mset::prepare(service::storage_proxy& proxy, request&& req) {
+    std::map<bytes, bytes> data_map;
+    if (req.arguments_size() == 0 || req.arguments_size() % 2 != 0) {
+        throw wrong_number_of_arguments_exception(req._command);
+    }
+
+    for (unsigned i = 0; i < req.arguments_size() / 2; i++) {
+        data_map.insert(std::make_pair(req._args[2 * i], req._args[2 * i + 1]));
```

Comment:
`data_map.emplace(req._args[2 * i], req._args[2 * i + 1]);`

Reply:
Also I still think this can be a std::vector<std::pair<bytes,bytes>> (or std::list or std::dequeue similar), there is no reason for this to be a std::map, which does needless work of sorting these parameters (which you don't need, unless I'm misunderstanding something).

Reply:
Agree.